preseed/provision.erb

```diff
@@ -131,8 +105,8 @@ d-i apt-setup/local<%= repos %>/key string http://keyserver.ubuntu.com/pks/looku
 # Install minimal task set (see tasksel --task-packages minimal)
 tasksel tasksel/first multiselect minimal, ssh-server, openssh-server
 
-<% if puppet_enabled %>
-<% puppet_package = @host.param_true?('enable-puppetlabs-pc1-repo') ? 'puppet-agent' : 'puppet' -%>
+<% if puppet_enabled && !(@host.param_true?('enable-puppetlabs-repo') || @host.param_true?('enable-puppetlabs-pc1-repo')) %>
+<% puppet_package = 'puppet' -%>
```

Comment:
If puppet is enabled and no puppetlabs repo is enabled, then the native OS package should be used (in whatever version the OS is bringing). In the finish script, there will be a redundant "apt-get install puppet" in that case, but that should be harmless.

If puppet is enabled and any PL repo is enabled, puppet (puppet3) or puppet-agent (puppet4) will be installed .

At least that's what I had in mind.